Mathlib/GroupTheory/Perm/Cycle/PossibleTypes.lean

```diff
@@ -0,0 +1,257 @@
+/-
+Copyright (c) 2023 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.GroupTheory.Perm.Cycle.Concrete
+
+/-! # Possible cycle types of permutations
+
+* For `m : Multiset ℕ`, `Equiv.Perm.withCycleType_nonempty_iff m`
+  proves that there are permutations with cycleType `m` if and only if
+  its sum is at most `Fintype.card α` and its members are at least 2.
+
+To prove this, we add some lemmas about lists which should probably be
+put elsewhere
+
+* `List.disjoint_map` and `List.disjoint_pmap` prove that
+  the images of disjoint lists under an injective (partial) map
+  are disjoint if that map is injective
+* Given `l : List ℕ`, `List.ranges` construct a list of lists of integers
+  whose zip is `[0,1,2,…]` and whose lengths are given by `l`.
+  For example, `List.ranges [2,0,3] = [[0,1], [], [2,3,4]]`.
+* `List.ranges_disjoint`, `List.ranges_nodup`, `List.ranges_lt`,
+  `List.ranges_lengths` prove properties of that function
+* Given `l : List ℕ` and a type `α` with more than `l.sum` elements,
+  `List.exists_pw_disjoint_with_card α l` constructs a `List (List α)`
+  with given lengths, pairwise disjoint members, each of them having no duplicates.
+-/
+
+
+/- section List
```

Comment:
Is this supposed to be commented out?

Reply:
Yes. It has been deplaced and will be deleted once it works there.